test/smoke/drive.js

```diff
@@ -28,32 +28,37 @@ async function sendEcho(ip) {
   });
 }
 
+const allHosts = inventory.masternodes.hosts.concat(inventory.seed_nodes.hosts);
+
 describe('Drive', () => {
-  describe('seed-1', () => {
-    it('should listen for ABCI connection', async function it() {
-      if (!variables.evo_services) {
-        this.skip('Evolution services are not enabled');
+  describe('All nodes', () => {
+    const echoInfo = {};
 
-        return;
-      }
+    before('Collect echo info from Drive', () => {
+      const promises = [];
 
-      // eslint-disable-next-line no-underscore-dangle
-      await sendEcho(inventory._meta.hostvars['seed-1'].public_ip);
-    });
-  });
+      for (const hostName of allHosts) {
+        // eslint-disable-next-line no-underscore-dangle
+        const requestEchoPromise = sendEcho(inventory._meta.hostvars[hostName].public_ip)
+          .then(() => {
+            echoInfo[hostName] = true;
+          })
+          .catch((e) => {
+            echoInfo[hostName] = e;
+          });
 
-  for (const hostName of inventory.masternodes.hosts) {
-    describe(hostName, () => {
-      it('should listen for ABCI connection', async function it() {
-        if (!variables.evo_services) {
-          this.skip('Evolution services are not enabled');
+        promises.push(requestEchoPromise);
+      }
 
-          return;
-        }
+      return Promise.all(promises).catch(() => Promise.resolve());
+    });
 
-        // eslint-disable-next-line no-underscore-dangle
-        await sendEcho(inventory._meta.hostvars[hostName].public_ip);
+    for (const hostName of allHosts) {
+      describe(hostName, () => {
+        it('should listen for ABCI connection', () => {
+          expect(echoInfo[hostName]).to.be.true();
```

Comment:
Will we see the error in this case?

Reply:
Yes, when drive fails to respond to the ping we get:
```
  1) Smoke
       Drive
         All nodes
           masternode-2
             should listen for ABCI connection:
     AssertionError: expected [Error: Can't connect to ABCI port: timeout.] to be true
      at Context.<anonymous> (test/smoke/drive.js:59:48)
      at processImmediate (node:internal/timers:466:21)
```

Is that good enough? Or should I collect errors in a different object?

Reply:
I would recommend using `expect.fail` as a more idiomatic way.